sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -317,7 +363,74 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
       }
       Aggregate(groupByAttrs, patchedAggExpressions, firstAggregate)
     } else {
```

Comment:
Can you check more here just like `} else if (distinct group size == 1 && the group has a filter) {`?

Reply:
Also, you need to update the comment: https://github.com/apache/spark/pull/27058/files#diff-29e82df7487a97f879691c1b525709aeR223

Reply:
I will add check like` } else if (distinctAggGroups.size == 1) {`
But I have checked whether the aggregation has a filter in the following places
https://github.com/apache/spark/blob/a4fd143fa92cb569764b669e5c53f6dfd1f1ae6a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala#L367

Reply:
You mean you cannot check that here? If agg funcs have no filter, we do nothing in this rule.

Reply:
Yes. I know. 
I have checked whether the aggregation has a filter.
https://github.com/apache/spark/blob/a4fd143fa92cb569764b669e5c53f6dfd1f1ae6a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala#L367
You means I should advance the check?

Reply:
yea, this is a nit comment though, I think 
```
    } else {
      // not need this `parition` computation when having no filter
      val (distinctAggExpressions, regularAggExpressions) = aggExpressions.partition(_.isDistinct)
      if (distinctAggExpressions.exists(_.filter.isDefined)) {
      ...
```